keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,644 @@
 
-## Grant Bucket Access
-This workflow describes the automation supporting granting access to an existing backend bucket.
+More information about COSI gRPC API is [here](#cosi-grpc-api)
 
-Here is the workflow:
+```
+    COSI Driver (s3.amazonaws.com)
+    |------------------------------------------|
+    | grpc ProvisionerGrantBucketAccess({      |
+    |      "name": "bar-$uuid",                |
+    |      "bucketID": "br-$uuid",             |
+    |      "parameters": {                     |
+    |         "key": "value"                   |
+    |      }                                   |
+    | })                                       |
+    |------------------------------------------|
+```
+Each BucketAccess is meant to map to one account (or its equivalent) in the OSP, and grant access for that account to access the bucket. Once the credentials are successfully generated in the OSP, the credential values are stored in a secret. The secret resides in the namespace of the driver. Since each driver is expected to run in its own namespace, the secret containing access credentials cannot be accessed directly by users.
 
-+ The admin creates a cluster-scoped `Bucket` representing the backend bucket.
-+ The user creates a `BucketRequest` referencing this `Bucket` instance and COSI follows the "Create Bucket" workflow.
-+ The user creates a `BucketAccessRequest` (BAR) which references their BR.
-+ COSI central controller sees a new `BucketAccessRequest` (BAR) and gets the `BAR.BucketAccessClass` (BAC).
-+ COSI creates the cluster-scoped `BucketAccess` (BA) filling in the field from the BAR and BAC.
-+ The sidecar sees the new BA and gRPC calls the driver to grant access to the backend bucket.
-+ COSI updates the BAR and BA to reference each other.
-+ The cosi-node-adapter, which has been waiting for the BAR to be processed, writes access tokens to the app pod's specified mount point.
+The sidecar sets `accessGranted=true` in the BucketAccess object and sets the `credentialsSecret` field to the name and namespace of the generated secret. At this point, the BucketAccess is ready to be utilized by workloads.
 
-## Revoke Bucket Access
-This workflow describes the automation supporting revoking access to an existing backend bucket, and the deletion of the cluster-scoped `BucketAccess` instance.
+## Bucket Provisioning (for pods)
 
-Here is the workflow:
+The following stakeholders are involved in the lifecycle of provisioning bucket for pods:
 
-+ COSI central controller sees the `BucketAccessRequest` (BAR) has been deleted.
-+ COSI gets the `BAR.BucketAccess` (BA).
-+ COSI deletes the BA, but finalizers prevent it from being garbage collected.
-+ The sidecar sees the BA has been deleted and gRPC calls the driver to revoke access to the backend bucket.
-+ COSI removes the BAR and BA finalizers and these instances are garbage collected.
+ - Users  - specify bucket in the pod definition
```

Comment:
>  - Users  - specify bucket in the pod definition

Is this intentionally generic? Do you want to say "...specify the bucket access request name in the pod definition"?

Reply:
This is a high level description of the user action. The details are provided below anyways